test/e2e/adminapi_resources.go

```diff
@@ -35,16 +36,15 @@ var _ = Describe("[Admin API] List Azure resources action", func() {
 		expectedResources, err := clients.Resources.ListByResourceGroup(ctx, clusterResourceGroup, "", "", nil)
 		Expect(err).NotTo(HaveOccurred())
 
+		expectedResourceIDs := make([]string, 0, len(expectedResources)+1)
+		for _, r := range expectedResources {
+			expectedResourceIDs = append(expectedResourceIDs, strings.ToLower(*r.ID))
+		}
+
 		By("adding VNet to list of valid Azure resource IDs")
-		aroResourceGroup := strings.TrimPrefix(clusterResourceGroup, "aro-")
-		vNet, err := clients.Resources.ListByResourceGroup(ctx, aroResourceGroup, "resourceType eq 'Microsoft.Network/virtualNetworks'", "", nil)
```

Comment:
@nilsanderselde (cc @mjudeikis) both of these lines are leaky - we don't need to make the aro- prefix assumption, or to assume there is a single vnet in that RG.  Both of these things might change when we start testing private clusters.